torch/csrc/autograd/functions/init.cpp

```diff
@@ -62,27 +63,89 @@ struct NoCtor {
 };
 
 template<typename C, typename T>
-static void addClass(PyObject* module, PyTypeObject& type, const char* name)
+static void addClass(PyObject* module, PyTypeObject& type, const char* name,
+  PyGetSetDef* function_properties=NULL, PyMethodDef* function_methods=NULL)
 {
-  createForwardFunctionPyTypeObject<T>(type, name);
+  createForwardFunctionPyTypeObject<T>(type, name, function_properties, function_methods);
   Py_INCREF(&type);
   PyModule_AddObject(module, name, (PyObject*)&type);
   registerCppFunction(typeid(C), &type);
 }
 
+template<typename T, typename ValueT, typename ParamsT, ValueT ParamsT::*ptr,
+         typename ConvertArgT, PyObject* (*Convert)(ConvertArgT)>
+PyObject* getTupleAttr(PyObject* obj, void* _unused)
+{
+  HANDLE_TH_ERRORS
+  THPCppFunction* self = (THPCppFunction*)obj;
+  auto& arr = (T*)(self->cdata.get())->*ptr;
+  auto num_elems = arr.size();
+  THPObjectPtr py_tuple = PyTuple_New(num_elems);
+  if (!py_tuple) return NULL;
+  for (size_t i = 0; i < num_elems; ++i) {
+    PyTuple_SET_ITEM(py_tuple.get(), i, Convert(arr[i]));
+  }
+  return py_tuple.release();
+  END_HANDLE_TH_ERRORS
+}
+
+template<typename T, typename ValueT, typename ParamsT, ValueT ParamsT::*ptr,
+         typename ConvertArgT, PyObject* (*Convert)(ConvertArgT)>
+PyObject* getValueAttr(PyObject* obj, void* _unused)
+{
+  HANDLE_TH_ERRORS
+  THPCppFunction* self = (THPCppFunction*)obj;
+  auto& val = (T*)(self->cdata.get())->*ptr;
+  return Convert(val);
+  END_HANDLE_TH_ERRORS
+}
+
+static struct PyGetSetDef conv_forward_properties[] = {
+  THP_FUNCTION_DEFAULT_PROPERTIES,
+  {(char*)"stride", (getter)getTupleAttr<ConvForward, std::vector<int>, ConvParams, 
+                                         &ConvParams::stride, long, PyInt_FromLong>, NULL, NULL, NULL},
+  {(char*)"padding", (getter)getTupleAttr<ConvForward, std::vector<int>, ConvParams, 
+                                         &ConvParams::padding, long, PyInt_FromLong>, NULL, NULL, NULL},
+  {(char*)"dilation", (getter)getTupleAttr<ConvForward, std::vector<int>, ConvParams, 
+                                         &ConvParams::dilation, long, PyInt_FromLong>, NULL, NULL, NULL},
+  {(char*)"transposed", (getter)getValueAttr<ConvForward, bool, ConvParams, 
+                                         &ConvParams::transposed, long, PyBool_FromLong>, NULL, NULL, NULL},
+  {(char*)"output_padding", (getter)getTupleAttr<ConvForward, std::vector<int>, ConvParams, 
+                                         &ConvParams::output_padding, long, PyInt_FromLong>, NULL, NULL, NULL},
```

Comment:
I originally avoided the ParamsT template argument and just went with &ConvForward::output_padding (same for the other params), but the compiler was complaining that the member pointer template parameter was specified incorrectly (I can reconstruct the error if you'd like).
Maybe it's a corner case (member pointer template parameter from a multiple inheritance parent), but the only way I could make it work was to explicitly reference the member in the context of the class it was defined in (ConvParams).